packages/compass-import-export/src/utils/collection-stream.js

```diff
@@ -69,57 +71,85 @@ class WritableCollectionStream extends Writable {
     this._executeBatch(callback);
   }
 
-  _executeBatch(callback) {
+  async _executeBatch(callback) {
     const documents = this.batch;
 
     this.batch = [];
 
-    this._collection().bulkWrite(
-      documents.map((doc) => ({ insertOne: doc })),
-      {
-        ordered: this.stopOnErrors,
-        retryWrites: false,
-        checkKeys: false,
-      },
-      (err, res) => {
+    const bulkWrite = promisify(
+      this._collection().bulkWrite.bind(this._collection())
+    );
+    let result;
+    let error;
+
+    try {
+      result = await bulkWrite(
+        documents.map((doc) => ({ insertOne: doc })),
+        {
+          ordered: this.stopOnErrors,
+          retryWrites: false,
+          checkKeys: false,
+        }
+      );
+    } catch (bulkWriteError) {
+      // Currently, the server does not support batched inserts for FLE2:
+      // https://jira.mongodb.org/browse/SERVER-66315
+      // We check for this specific error and re-try inserting documents one by one.
+      if (bulkWriteError.code === 6371202) {
```

Comment:
Do we want to also do `this.BATCH_SIZE = 1;` in this case, to reduce the number of times that we run into this condition? (The `BATCH_SIZE` check above would need to be turned into `>=` instead of `===` in that case, but as you mentioned yesterday, that might be more intuitive anyway)